### PR TITLE
chore: add script to create adaptor packaging artifact

### PR DESCRIPTION
**Pull request by joel-wong-aws**
### What was the problem/requirement? (What/Why)

Create a script to produce the adaptor packaging artifact for KeyShot, similar to those for [Blender](https://github.com/aws-deadline/deadline-cloud-for-blender/blob/mainline/scripts/create_adaptor_packaging_artifact.sh), [Houdini](https://github.com/aws-deadline/deadline-cloud-for-houdini/blob/mainline/scripts/create_adaptor_packaging_artifact.sh), and [Nuke](https://github.com/aws-deadline/deadline-cloud-for-nuke/blob/mainline/scripts/create_adaptor_packaging_artifact.sh)

### Was this change documented?

No, N/A

### Is this a breaking change?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: scripts/create_adaptor_packaging_artifact.sh [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+if [ $SOURCE = 1 ]; then
+    # In source mode, openjd-adaptor-runtime-for-python must be alongside this adaptor source
+    RUNTIME_INSTALLABLE=$SCRIPTDIR/../../openjd-adaptor-runtime-for-python
+    CLIENT_INSTALLABLE=$SCRIPTDIR/../../deadline-cloud
+    ADAPTOR_INSTALLABLE=$SCRIPTDIR/..
+
+    if [ "$CONDA_PLATFORM" = "win-64" ]; then
+        DEPS="pyyaml jsonschema pywin32"
+    else
+        DEPS="pyyaml jsonschema"
```
Comment (crowecawcaw) [inline scripts/create_adaptor_packaging_artifact.sh:78]: Does python not need to be installed for Linux and Mac?

File: scripts/create_adaptor_packaging_artifact.sh [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    --source) SOURCE=1 ; shift ;;
+    --platform) CONDA_PLATFORM="$2" ; shift 2 ;;
+    --python) PYTHON_VERSION="$2" ; shift 2 ;;
+    --tar-base) TAR_BASE="$2" ; shift 2 ;;
+    *) echo "Unexpected option: $1"; exit 1 ;;
+  esac
+done
+
+if [ "$CONDA_PLATFORM" = "linux-64" ]; then
+    PYPI_PLATFORM=manylinux2014_x86_64
```
Comment (crowecawcaw) [inline scripts/create_adaptor_packaging_artifact.sh:31]: How did you choose this platform and the Mac on? They look outdated but I'd guess it's required for some reason.

**Review by crowecawcaw — COMMENTED** [review]
How were you able to test this? Did you test it on multiple platforms?

File: scripts/create_adaptor_packaging_artifact.sh (hunk 2 above)
Comment (mwiebe) [reply scripts/create_adaptor_packaging_artifact.sh:31]: Here's a list of the manylinux options: https://github.com/pypa/manylinux
I couldn't find docs for the mac platforms

File: scripts/create_adaptor_packaging_artifact.sh (hunk 1 above)
Comment (joel-wong-aws) [reply scripts/create_adaptor_packaging_artifact.sh:78]: `pywin32`, which provides utilities for Windows, doesn't have to be 🙂

File: scripts/create_adaptor_packaging_artifact.sh (hunk 2 above)
Comment (joel-wong-aws) [reply scripts/create_adaptor_packaging_artifact.sh:31]: I used the same platforms as all the other packaging artifact files; not sure exactly which this was chosen for them but I wanted to stay consistent

**Comment by joel-wong-aws** [comment]
Closing in favour of https://github.com/aws-deadline/deadline-cloud-for-keyshot/pull/109

**Pull request closed without merging**